src/foam/nanos/crunch/ServerCrunchService.java

```diff
@@ -224,51 +224,32 @@ public UserCapabilityJunction getJunctionForSubject(
       Predicate associationPredicate = getAssociationPredicate_(x);
 
       // Check if a ucj implies the subject.realUser has this permission in relation to the user
-      UserCapabilityJunction ucj = (UserCapabilityJunction)
+      var ucj = (UserCapabilityJunction)
         userCapabilityJunctionDAO.find(AND(associationPredicate,targetPredicate));
-      if ( ucj != null ) {
-        return ucj;
+      if ( ucj == null ) {
+        ucj = buildAssociatedUCJ(x, capabilityId, subject);
+        ucj.setStatus(CapabilityJunctionStatus.AVAILABLE);
```

Comment:
I think it makes sense to have the buildAssociatedUCJ function set the ucj status to AVAILABLE. Then can avoid setting it after the calling code.

Reply:
Good catch, I didn't think of that